scene/gui/text_edit.cpp

```diff
@@ -7259,7 +7259,7 @@ void TextEdit::_update_selection_mode_word() {
 	int end = beg;
 	PackedInt32Array words = TS->shaped_text_get_word_breaks(text.get_line_data(line)->get_rid());
 	for (int i = 0; i < words.size(); i = i + 2) {
-		if ((words[i] < caret_pos && words[i + 1] > caret_pos) || (i == words.size() - 2 && caret_pos == words[i + 1])) {
+		if ((words[i] < caret_pos && words[i + 1] >= caret_pos) || (i == words.size() - 2 && caret_pos == words[i + 1])) {
```

Comment:
This works when the right side of the word is double clicked, but not the left side.

Reply:
I will adjust the condition to fix this.